cb/cb.py

```diff
@@ -80,9 +80,10 @@ def __call__(self, *args):
             if select_as_int in range(len(inputs)):
                 self.out = args[select_as_int]
                 return self.out
-            # TODO(raj): Handle the case where we select the constant value or
-            # resort to default.
-            raise Exception()
+            if has_constant:
```

Comment:
This logic is actually not accurate. It should be

```
if has_constant:
    return __get_config_bits(mux_sel_bits, mux_sel_bits + width)

return BitVector(0, width)
```

Reply:
That's equivalent to the implementation, but does that imply that the default value can change via configuration? Why is the default value a generator parameter then (rather than a configuration setting)? This is more of a high level design question, I'll update it to match the implementation, but we should consider what the specificaiton should be..

Reply:
The nomenclature here is a bit confusing. There is a "constant value" configuration register. This value is set at configuration time. This value is also used as a mux input. "default value" refers to what value this register should take upon reset. "default value" is therefore a generator parameter. It's kind of like:

```
ConfigReg constant_value
if reset:
    constant_value <= default_value
...
if config_addr == <some address>:
    constant_value <= config_data

out = Mux(inputs + [constant_value], select)
```

"default value" should really be renamed to something like "constant value reset value" (clunky).

Reply:
none, will approve changes.